### PR TITLE
(fix) Increased rate limit to 100 req/min & Added pagination to thread-service

### PR DESCRIPTION
**Pull request by sarunnut1236**
Pagination includes "getAllThreads" and "searchThreads"

### CODE REVIEW

File: proto/thread.proto
Comment (sarunnut1236) [inline proto/thread.proto:1]: ทำ REST ไว้ แต่ว่าไม่ได้ใช้ เลยลบทิ้ง

File: api-gateway/src/utils/sercurity.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import hpp from "hpp";
 
 // Rate limiting configuration
 const limiter = rateLimit({
-  windowMs: 15 * 60 * 1000, // 15 minutes
+  windowMs: 60 * 1000, // 1 minute
```
Comment (ThanatWongsamut) [inline api-gateway/src/utils/sercurity.ts:9]: Why we need to reduce this

File: api-gateway/src/controllers/thread-controller.ts [hunk 2]
```diff
@@ -76,7 +78,9 @@ export const deleteThread = controllerWrapper(
 export const searchThreads = controllerWrapper(
   async (req: Request, res: Response) => {
     const query = (req.query.query as string) || "";
-    const threads = await grpcRequest("searchThreads", { query });
+    const page = Number(req.query.page) || 0;
+    const pageSize = Number(req.query.pageSize) || 25;
```
Comment (ThanatWongsamut) [inline api-gateway/src/controllers/thread-controller.ts]: Maybe bound the maximum pageSize like if x > 100 ? 100 : x

**Review by ThanatWongsamut — COMMENTED** [review]
Left some comment

File: api-gateway/src/utils/sercurity.ts (hunk 1 above)
Comment (sarunnut1236) [reply api-gateway/src/utils/sercurity.ts:9]: We cause frontend some problem with our rate limiting; the same config was 100 req/15 min which is 7 req/min; so we need to increase window size

File: api-gateway/src/controllers/thread-controller.ts (hunk 2 above)
Comment (sarunnut1236) [reply api-gateway/src/controllers/thread-controller.ts]: Edited

**Review by ThanatWongsamut — APPROVED** [review]
Nice

**Pull request merged**